sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/LookupCatalog.scala

```diff
@@ -26,27 +28,31 @@ import org.apache.spark.sql.catalyst.TableIdentifier
 @Experimental
 trait LookupCatalog {
 
-  def lookupCatalog: Option[(String) => CatalogPlugin] = None
+  def lookupCatalog: Option[String => CatalogPlugin] = None
```

Comment:
Why not require this lookup function instead? I think it makes more sense to ensure it is always present.

Reply:
Like the idea. It will force the users to think about the correct lookup functions when using the interface. At the last resort, users can supply a dummy function that always returns NoSuchCatalogException.